job-server/test/spark.jobserver/LocalContextSupervisorSpec.scala

```diff
@@ -80,8 +82,8 @@ class LocalContextSupervisorSpec extends TestKit(LocalContextSupervisorSpec.syst
 
     it("should be able to add multiple new contexts") {
       supervisor ! AddContext("c1", contextConfig)
-      supervisor ! AddContext("c2", contextConfig)
       expectMsg(ContextInitialized)
+      supervisor ! AddContext("c2", contextConfig)
```

Comment:
The nonblocking way is to add a queue and have add context requests wait on the queue, only take things off queue if another context is done initializing. 

-Evan
"Never doubt that a small group of thoughtful, committed citizens can change the world" - M. Mead

> On Jan 18, 2015, at 6:41 AM, Fernando Otero <notifications@github.com> wrote:
> 
> In job-server/test/spark.jobserver/LocalContextSupervisorSpec.scala:
> 
> >        expectMsg(ContextInitialized)
> > +      supervisor ! AddContext("c2", contextConfig)
> I took a deeper look at the code, and since JobServer is the only one that spawns a LocalContextSupervisorActor then using await on startcontext seems safe.
> …
> —
> Reply to this email directly or view it on GitHub.
>